catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/view/Segment.view.js

```diff
@@ -0,0 +1,183 @@
+                this.formSegments.show(new Segment.SegmentCollectionView({
+                    model: this.model,
+                    collection: this.model.get('segments'),
+                    showHeader: knownFields.length > 0 ? false : true
+                }));
+                if (FieldDescriptors.isCustomizableSegment(this.model.get('segmentType'))) {
+                    this.customizable.show(new Customizable.CustomizableCollectionView({
+                        model: this.model
+                    }));
+                    if (this.model.get('fields').models.length > 0) {
+                        var associationModel = this.model.get('associationModel');
+                        this.associations.show(new Association.AssociationCollectionView({
+                            parentId: this.model.get("segmentId"),
+                            model: associationModel,
+                            collection: new Backbone.Collection(associationModel.getAssociationsForId(this.model.get('segmentId')))
+                        }));
+                    }
+                }
+                this.setupPopOvers();
+            },
+            removeSegment: function () {
+                wreqr.vent.trigger('removeSegment:' + this.model.get('parentId'), this.model.get('segmentId'));
+            },
+            addSegment: function () {
+                this.model.addSegment();
+                this.render();
+            },
+            updateTitle: function () {
+                if (!this.model.get('multiValued') && this.model.constructTitle) {
+                    var title = this.$('.segment-title-'+this.model.get('simpleId'));
+                    title.html(this.model.constructTitle());
+                }
+            },
+            serializeData: function () {
+                var data = {};
+
+                if (this.model) {
+                    data = this.model.toJSON();
+                }
+                data.editableSegment = this.editableSegment;
+                if (!this.model.get('multiValued') && this.model.constructTitle) {
+                    data.title = this.model.constructTitle();
+                } else {
+                    data.title = this.model.get('segmentName');
+                }
+                data.customizable = FieldDescriptors.isCustomizableSegment(this.model.get('segmentType')) && !this.model.get('multiValued');
+                data.showFields = this.model.get('fields').models.length > 0;
+                data.showAssociations = FieldDescriptors.isCustomizableSegment(this.model.get('segmentType')) && this.model.get('fields').models.length > 0;
+                return data;
+            },
+            /**
+             * Set up the popovers based on if the selector has a description.
+             */
+            setupPopOvers: function () {
+                var view = this;
+                this.model.get('fields').models.forEach(function (each) {
```

Comment:
Not sure why we have the check in serializeData. I removed it. Changed the models.forEach to each since it is a backbone collection